lib/Conversion/ExportVerilog/ExportVerilog.cpp

```diff
@@ -2270,17 +2270,22 @@ SubExprInfo ExprEmitter::visitUnhandledExpr(Operation *op) {
 // NameCollector
 //===----------------------------------------------------------------------===//
 
-
-static ConstantOp isSingleConstantAssign(Operation *op) {
+static std::pair<ConstantOp, AssignOp> isSingleConstantAssign(Operation *op) {
   auto wire = dyn_cast<WireOp>(op);
   if (!wire)
     return {};
-  if (!wire->hasOneUse())
-    return {};
```

Comment:
This looks suspiciously like it would always return `{}` if anything ever read from that wire (which would contribute a second use next to the assign)... :grimacing:

Reply:
it was mainly catching all the dangling wires and cleaning them up.  The right thing to do is to get rid of the wire pre-emission and eagerly fold in any assign whose inputs are available.